### DISCUSSION TITLE
Why would nuclei send requests when there's no target provided?

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [projectdiscovery/nuclei#2410] [Q&A] [answered]
I started nuclei with no arguments, and it hangs.

So I started it again with `-debug`, and it is sending a lot of requests:

```
# nuclei -debug

...
[DBG] [api-github] Dumped HTTP response https://api.github.com/user

HTTP/1.1 401 Unauthorized
Connection: close
Content-Length: 141
...
```

What is nuclei doing?

**Comment by ehsandeep** [comment, marked as answer]
@CaledoniaProject nuclei as default load all the templates to run including self-contained [token-spray](https://github.com/projectdiscovery/nuclei-templates/tree/master/token-spray) templates which doesn't requires target input as it's already included in the template.

**Reply by CaledoniaProject** [reply]
This is weird, I think nuclei should change the default behavior. The reason is quite simple: I didn't provide any arguments to token-spray templates, it is useless to run them.

**Reply by ehsandeep** [reply]
@CaledoniaProject it was due to bug that let some template to run and now being tracked here - https://github.com/projectdiscovery/nuclei/issues/2412, thanks for flagging this.

**Reply by CaledoniaProject** [reply]
Can you also take a look at this bug? https://github.com/projectdiscovery/nuclei/issues/2400

**Comment by ehsandeep** [comment]
@CaledoniaProject this issue should be fixed with the latest release.
